jwst/wfss_contam/wfss_contam.py

```diff
@@ -106,15 +108,24 @@ def contam_corr(input_model, waverange, photom, max_cores):
 
     # Initialize the simulated image object
     simul_all = None
-    obs = Observation(image_names, seg_model, grism_wcs, filter_name,
+    obs = Observation(image_names, seg_model, grism_wcs_dict, filter_name,
                       boundaries=[0, 2047, 0, 2047], max_cpu=ncpus)
 
+    # Create dict of offsets to pass to disperse_all to create simulated image
+    # with sources in correct locations
+    offset_dict = dict()
+    for slit in output_model.slits:
+        source_entry = offset_dict.get(int(slit.source_id), dict())
+        # Add entry in source dict for each order
+        source_entry[slit.meta.wcsinfo.spectral_order] = (slit.xstart - 1, slit.ystart - 1)
+        offset_dict[int(slit.source_id)] = source_entry
```

Comment:
If there's more than one slit per source_id (i.e. multiple orders), will the dictionary have redundant entries and/or one entry overwriting another? Or is there another "dimension" to the dictionary entry for a given source_id that contains the unique data for each order?

Reply:
I could add a comment here, but the dict.get() method will pull the entry for the `source_id` if it exists, or return an empty `dict()` if it is not yet in `offset_dict.keys()`. The next lines add a new entry to the dict and then assign the updated dict to the parent `offset_dict`. Not the cleanest, but it worked best with the code in `wfss_contam.observations.Observation.disperse_all`, where the simulated images are built by order - in this way, it checks the presence of a source, then checks on order before simulating the source into the full frame.